### ISSUE TITLE
App keeps stopping on react 0.58.5

### ISSUE THREAD

**Issue by jeronimo13** [skrafft/react-native-jitsi-meet#22]
Hi
Many thanks for tremendous work on this project!
I have a react native app which I already connected to iOS and it works perfectly on both iPhone and iPad
But with Android, I have problems with integration. I’ve tried 2 times to add it to my project as manual says and came up with app crashing when I call `initiateVideoCall`.

I’ve even created a brand new project with only one purpose: add react-native-jitsi-meet dependency to the project. 

Disclaimer: I'm new to the react native world. I've come up here from Java backend world, so maybe there are some obvious things I missed.

During 3 tries this is what I always bumped to after doing doing readme installation and re-running react-native run-android over and over again:

First **error**:
```
> Task :app:preDebugBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:preDebugBuild'.
> Android dependency 'com.facebook.react:react-native' has different version for the compile (0.58.5) and runtime (0.57.8) classpath. You should manually set the same version via DependencyResolution

```

My solution: in app/build.gradle:
dependencies {
    implementation project(':react-native-gesture-handler')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
    implementation "com.facebook.react:react-native:0.58.5"  // **changed this to 0.58.5** 
    implementation(project(':react-native-jitsi-meet'))
}

Second error:
```
> Task :app:processDebugManifest FAILED
/Users/herman/Documents/work/BoozerApp/android/app/src/main/AndroidManifest.xml:24:7-34 Error:
        Attribute application@allowBackup value=(false) from AndroidManifest.xml:24:7-34
        is also present at [org.jitsi.react:jitsi-meet-sdk:1.21.0] AndroidManifest.xml:31:9-35 value=(true).
        Suggestion: add 'tools:replace="android:allowBackup"' to <application> element at AndroidManifest.xml:19:5-38:19 to override.
/Users/herman/Documents/work/BoozerApp/android/app/src/main/AndroidManifest.xml Error:
        uses-sdk:minSdkVersion 16 cannot be smaller than version 21 declared in library [org.jitsi.react:jitsi-meet-sdk:1.21.0] /Users/herman/.gradle/caches/transforms-1/files-1.1/jitsi-meet-sdk-1.21.0.aar/68ab6f17f2b7329fac831cb6388f136c/AndroidManifest.xml as the library might be using APIs not available in 16
        Suggestion: use a compatible library with a minSdk of at most 16,
                or increase this project's minSdk version to at least 21,
                or use tools:overrideLibrary="org.jitsi.meet.sdk" to force usage (may lead to runtime failures)

See http://g.co/androidstudio/manifest-merger for more information about the manifest merger.
```

My solution, in build.gradle change
`        minSdkVersion = 21`

Next error:
```
> Task :app:processDebugManifest FAILED
/Users/herman/Documents/work/BoozerApp/android/app/src/main/AndroidManifest.xml:24:7-34 Error:
        Attribute application@allowBackup value=(false) from AndroidManifest.xml:24:7-34
        is also present at [org.jitsi.react:jitsi-meet-sdk:1.21.0] AndroidManifest.xml:31:9-35 value=(true).
        Suggestion: add 'tools:replace="android:allowBackup"' to <application> element at AndroidManifest.xml:19:5-38:19 to override.
```

Solution is to change `AndroidManifest`:
```
    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false" //<---------------------- I removed this line
      android:theme="@style/AppTheme">
```

This one is pretty simple, a compilation error for @Nullable annotation:
```
> Task :app:compileDebugJavaWithJavac FAILED
/Users/herman/Documents/work/BoozerApp/android/app/src/main/java/com/boozerbar/MainApplication.java:39: error: cannot find symbol
    protected @Nullable String getBundleAssetName() {
               ^
  symbol: class Nullable
1 error
```

In MainApplication.java added `import android.support.annotation.Nullable;`. I also tried with  `import javax.annotation.Nullable;` no difference

This one has pretty big stacktrace, but in the end it says something about Java 8:
```
> Task :app:transformClassesWithDexBuilderForDebug
com.android.builder.dexing.DexArchiveBuilderException: com.android.builder.dexing.DexArchiveBuilderException: Failed to process /Users/herman/.gradle/caches/transforms-1/files-1.1/react-native-webrtc-1.67.1-jitsi-9.aar/225980d4b2d9d17294dc212a2fcf0eb3/jars/classes.jar
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488)
        at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
        at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
        at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:722)
        at com.android.ide.common.internal.WaitableExecutor.waitForTasksWithQuickFail(WaitableExecutor.java:146)
        at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.transform(DexArchiveBuilderTransform.java:405)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:239)
        at com.android.build.gradle.internal.pipeline.TransformTask$2.call(TransformTask.java:235)
        at com.android.builder.profile.ThreadRecorder.record(ThreadRecorder.java:102)
        at com.android.build.gradle.internal.pipeline.TransformTask.transform(TransformTask.java:230)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:317)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:309)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:185)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:97)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:596)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
        at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: com.android.builder.dexing.DexArchiveBuilderException: Failed to process /Users/herman/.gradle/caches/transforms-1/files-1.1/react-native-webrtc-1.67.1-jitsi-9.aar/225980d4b2d9d17294dc212a2fcf0eb3/jars/classes.jar
        at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.launchProcessing(DexArchiveBuilderTransform.java:900)
        at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.lambda$convertToDexArchive$6(DexArchiveBuilderTransform.java:825)
        at java.base/java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1448)
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1603)
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: com.android.builder.dexing.DexArchiveBuilderException: Error while dexing.
The dependency contains Java 8 bytecode. Please enable desugaring by adding the following to build.gradle
android {
    compileOptions {
        sourceCompatibility 1.8
        targetCompatibility 1.8
    }
}
See https://developer.android.com/studio/write/java8-support.html for details. Alternatively, increase the minSdkVersion to 26 or above.

        at com.android.builder.dexing.D8DexArchiveBuilder.getExceptionToRethrow(D8DexArchiveBuilder.java:124)
        at com.android.builder.dexing.D8DexArchiveBuilder.convert(D8DexArchiveBuilder.java:101)
        at com.android.build.gradle.internal.transforms.DexArchiveBuilderTransform.launchProcessing(DexArchiveBuilderTransform.java:895)
        ... 5 more
Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete
        at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:65)
        at com.android.tools.r8.utils.ExceptionUtils.withD8CompilationHandler(ExceptionUtils.java:43)
        at com.android.tools.r8.D8.run(D8.java:90)
        at com.android.builder.dexing.D8DexArchiveBuilder.convert(D8DexArchiveBuilder.java:99)
        ... 6 more
Caused by: com.android.tools.r8.utils.AbortException: Error: Invoke-customs are only supported starting with Android O (--min-api 26)
        at com.android.tools.r8.utils.Reporter.failIfPendingErrors(Reporter.java:116)
        at com.android.tools.r8.utils.Reporter.fatalError(Reporter.java:74)
        at com.android.tools.r8.utils.ExceptionUtils.withCompilationHandler(ExceptionUtils.java:59)
        ... 9 more


> Task :app:transformClassesWithDexBuilderForDebug FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithDexBuilderForDebug'.
> com.android.build.api.transform.TransformException: com.android.builder.dexing.DexArchiveBuilderException: com.android.builder.dexing.DexArchiveBuilderException: Failed to process /Users/herman/.gradle/caches/transforms-1/files-1.1/react-native-webrtc-1.67.1-jitsi-9.aar/225980d4b2d9d17294dc212a2fcf0eb3/jars/classes.jar
```

My solution is to add: 
```
    compileOptions {
        sourceCompatibility 1.8
        targetCompatibility 1.8
    }
```
to `android` section

This one is pretty mythical for me:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformNativeLibsWithMergeJniLibsForDebug'.
> More than one file was found with OS independent path 'lib/x86_64/libjsc.so'
```
so I've googled around and copy pasted from  [here](https://dev.to/sasurau4/tips-for-react-native-upgrade-to-v0584-197k )
Solution is to edit `android/app/build.gradle`

```
android {
    ...
    packagingOptions {
        pickFirst 'lib/x86_64/libjsc.so'
        pickFirst 'lib/arm64-v8a/libjsc.so'
    }
    ...
}

```

SO, after everything I've done I've come up with green build, app is up and running, but unfortunately, app is crashing with the message: 'jetsitest keeps stopping', when I call `initiateVideoCall`.
 I've tried to move `initiateVideoCall` function from `componentDidMount` to  `<Button title='Enter chat' onPress={()=> this.initiateVideoCall()}/>` with the same result, only difference is now I can crash app via button 🗡  👯‍♂️ 
I've also tried to do `./gradlew clean`, delete `node_modules` and do `yarn`. Also, cleared `~/.gradle/cache` folder a couple of times with all possible combinations.

The app is crashing when calling `initiateVideoCall` both on emulators(Pixel and Pixel 2 with Android 9) and on device (Sumsung galaxy s8 with (?)Android 8)

Logs are clear both in `adb logcat | grep 'jitsitest'` and remote debugger is clean

I have a react-native v. 0.58.5
```
jitsitest herman$ react-native info

  React Native Environment Info:
    System:
      OS: macOS 10.14.3
      CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
      Memory: 115.56 MB / 16.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 8.11.1 - /usr/local/bin/node
      Yarn: yarn install v0.27.5
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.65s. - /usr/local/bin/yarn
      npm: 6.0.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 12.1, macOS 10.14, tvOS 12.1, watchOS 5.1
    IDEs:
      Android Studio: 3.3 AI-182.5107.16.33.5264788
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.6.3 => 16.6.3 
      react-native: 0.58.5 => 0.58.5 
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7
```

I've posted that journey so maybe somebody can save some time or @skrafft could benefit from my issues and updates a readme, or maybe I'm just stupid and missing some obvios solution

TL;dr
After successful build app is crashing without any logs with error App keeps stoping.

@skrafft  thanks again for your work!

**Comment by jeronimo13** [comment]
Ok, I sleep a bit and resolve my issue

The problem was that I was using react-native 0.58.5, not 0.57.8. I downgrade to 0.57.8 and it works like a charm

If anybody still interested in upgrading to 0.58.5 you should look here:
[android: add @ReactModule annotation ](https://github.com/oney/react-native-webrtc/commit/8def7d0dc9370934a4d56770d63d742c2a203fea)
jitsi-meet-sdk using 1.67 version of webrtc, which is outdated. The current version is 1.69.0, where they fix this issue with annotation
I'm pretty sure the problem is in versioning, because when I debug Android native code there was an error `Could not find @ReactModule annotation in com.oney.WebRTCModule.WebRTCModule` which webrtc team fixed in 1.69.0 according to [this](https://github.com/oney/react-native-webrtc/issues/575)

TL;dr
don't use react-native 0.58.5 with jitsi-meet-sdk, because they use old version of webrtc. Use 0.57.8

**Comment by bhayward93** [comment]
Thanks for taking the time to share that @jeronimo13.

**Comment by gen-failure** [comment]
Was this somehow solved? Is there a version of react native officially supported?

**Comment by skrafft** [comment]
@gen-failure, I'm using it with RN 0.59.5 without the mentioned problem. Just be sure to be on the latest version of react-native-jitsi-meet.
